include/mbgl/shaders/mtl/heatmap_texture.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <mbgl/shaders/heatmap_texture_layer_ubo.hpp>
+#include <mbgl/shaders/shader_source.hpp>
+#include <mbgl/shaders/mtl/common.hpp>
+#include <mbgl/shaders/mtl/shader_program.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+template <>
+struct ShaderSource<BuiltIn::HeatmapTextureShader, gfx::Backend::Type::Metal> {
+    static constexpr auto name = "HeatmapTextureShader";
+    static constexpr auto vertexMainFunction = "vertexMain";
+    static constexpr auto fragmentMainFunction = "fragmentMain";
+
+    static constexpr AttributeInfo attributes[] = {
+        {0, gfx::AttributeDataType::Short2, 1, "a_pos"},
+    };
+    static constexpr UniformBlockInfo uniforms[] = {
+        MLN_MTL_UNIFORM_BLOCK(1, true, true, HeatmapTextureDrawableUBO),
+    };
+    static constexpr TextureInfo textures[] = {
+        {0, "u_image"},
+        {1, "u_color_ramp"},
+    };
+    static constexpr auto source = R"(
+
+struct VertexStage {
+    short2 pos [[attribute(0)]];
+};
+
+struct FragmentStage {
+    float4 position [[position, invariant]];
+    float2 pos;
+};
+
+struct alignas(16) HeatmapTextureDrawableUBO {
+    float4x4 matrix;
+    float2 world;
+    float opacity;
+    bool overdrawInspector;
```

Comment:
`overdrawInspector` is a mechanism used only with Metal. Usually this flag is in the `PermutationUBO`, but because I don't have this UBO for heatmap layer I put it at the end of `DrawableUBO`. There is no match for it in GLSL, GL uses this define `#ifdef OVERDRAW_INSPECTOR`.

Reply:
This is the UBO in GLSL:
```
layout (std140) uniform HeatmapTextureDrawableUBO {
    highp mat4 u_matrix;
    highp vec2 u_world;
    highp float u_opacity;
    lowp float pad0_;
};
```
and this is the UBO in Metal:
```
struct alignas(16) HeatmapTextureDrawableUBO {
    float4x4 matrix;
    float2 world;
    float opacity;
    bool overdrawInspector;
    uint8_t pad1, pad2, pad3;
};
```

Reply:
I see, that should be fine then I think. Just add a comment explaining you're re-using this padding for a Metal-only feature.

Reply:
Done, please check the comment.